### ISSUE TITLE
Only show emojis for `:emoji:`, not regular smileys

### ISSUE THREAD

**Issue by lucaswerkmeister** [gitterHQ/gitter#781]
This already came up in #601 (and CC #164), but I want to have a separate issue for it: The only way to produce emojis should be to actually spell out the emoji in colons, i. e., `:smile:`. Regular smileys, like `:)`, `:D`, `:/`, should be text, not emoji images.

(My main issue is with `:)` and `:/`, which look very different from their emoji counterparts :smile: and :pensive:. :smile: looks way too similar to :grinning:, and :pensive: looks a lot more sad than `:/`.)

**Comment by asmeurer** [comment]
:+1:

**Comment by quintesse** [comment]
:+1:

**Comment by ctrueden** [comment]
+1 ;-)

**Comment by MadLittleMods** [comment]
We may look to make this an option in the future :)

**Comment by asmeurer** [comment]
Wouldn't that just be a placebo? If I set it so that I see :) instead of :smile: but others still see :smile: then it's not really doing what I want. Or do you mean a per-room option?

**Comment by smathy** [comment]
Yes please, I'm really sick of seeing things like:  property :stuck_out_tongue_winking_eye:roperty_name

**Comment by vermiculus** [comment]
:+1:

**Comment by georgiel** [comment]
oh touché until now I didn't mind it, they're easier.

**Comment by quintesse** [comment]
> I think the :) should turn into :grinning::grinning: instead of :smile:

To me what I associate with `:)` is http://unicode.org/emoji/charts/full-emoji-list.html#263a , especially when looking at the most simplistic icons. In the icon set this is supposedly `:relaxed:` :relaxed: 
My association with those "open mouthed grins" is one of happiness, of exuberance.

So why not leave the typed smileys alone and add a special button for inserting emojis?
The same way that mobile applications like Whatsapp do it, if I type `:)` it stays that way and if I want to insert an emoji I use the button (this also makes it unnecessary to figure out what the emoji I want is called)

**Comment by lucaswerkmeister** [comment]
That button exists, it’s called the colon key ;)

(Seriously though, I think the selection menu you get from typing a colon – searchable by typing some more – is suffiicient.)

**Comment by quintesse** [comment]
I don't agree, if you have absolutely no idea how the emoji is called you can type all you want but you won't find it. I literally had to look up the emoji's name I put in my previous message on Google to be able to type it. Heck even GMail has a emoji selector!

**Comment by vermiculus** [comment]
I agree with precedent here; the colon is sufficient. For searching, there's <http://www.emoji-cheat-sheet.com>. It would be a duplicated effort to index them all within gitter again. If it's really necessary, it could link out to a static index.

**Comment by smathy** [comment]
Just do (as close as feasible to) what github does, seems obvious to me

**Comment by asmeurer** [comment]
If you are on OS X you can type CMD-CTRL-SPACE to pull up an emoji selector.

Anyway, an emoji selector is a separate issue. I don't care if there is one or isn't, just stop turning non-emoji into emoji!

**Comment by LB--** [comment]
+1, just make it work exactly like GitHub (isn't that the point?)

**Comment by abhisekp** [comment]
:-1: I don't agree with this. The `:)` is as much a smiley as `:smile:`. Both are same. Why see a text instead of an image? But I agree with that the smiley image must be modified to actually represent a smile instead of a laughing emoticon.

**Comment by ctrueden** [comment]
> Why see a text instead of an image?

Two reasons:

1. __Consistency.__ GitHub issues does not render `:)` as a smiley.
2. __Confusion.__ It is confusing when character sequences containing special characters, but which were not intended to represent emoticons, get overaggressively mangled that way. (See e.g. `property :property_name` example above.)

**Comment by abhisekp** [comment]
@ctrueden If someone wants to write something as a part of code or something related which they don't want to be rendered into an image, then they may wrap the text with two backticks \`\:property_name\` or they can put a `\` (backslash) and escape the character from parsing. 

Moreover, Github is not a messaging site. But gitter.im is and it should behave and work to some extend like any other IM services (though it's more programmer oriented).

**Comment by lucaswerkmeister** [comment]
The backticks are not always available, especially on mobile (and this is a messaging site, it should work well on mobile).

And not all IM services are equal. Automatic emojis might be appropriate for a casual chat app, but less so for a service where we want to have serious development discussions.

**Comment by smathy** [comment]
@abhisekp perfect, working like other chat apps would be great. All other group chat apps that I've used either do not emoji-fy ascii at all unless it's real colon-string-colon emoji, or they provide an option for automatically emoji-fying non-emoji. Either would be perfect for gitter.

**Comment by abhisekp** [comment]
@lucaswerkmeister Well, adding fun to "serious" dev discussion is fun too :smiley_cat: 

@smathy I agree with you about option to turn off the auto emoji-fy (emoji-fying by default).

**Comment by georgiel** [comment]
Instead of off by default or on by default.

How about a dialogue asking which you'd prefer on first login and the option in the settings for changing it later?

**Comment by asmeurer** [comment]
Some people here seem to be forgetting that you can still type emoji in gitter using `:smile:` or `:joy_cat:` or whatever. Those who know how to access them (such as users on mobile) could also add the literal Unicode characters as well. 

This issue is only about the automatic replacement of `:)` with :smile: and so on. `:)` could still itself be used as an emoticon (just not an emoji)! This doesn't prevent non-serious developer discussions.

**Comment by LB--** [comment]
@asmeurer +1, I don't want an emoji image unless I type the Unicode emoji character or the `:colon-name:` of the emoji.

**Comment by rphares** [comment]
An on/off option would be fantastic, and the : smile : is quite extreme, a closed-mouth smile would be fantastic for those not meaning to grin at everything said with a smile... :)

**Comment by flaviocopes** [comment]
:+1:

**Comment by Demon000** [comment]
The :smile: is too much of a grin. It should be replaced with more of a smile one, prototype: 
![1f604](https://cloud.githubusercontent.com/assets/12939434/11778488/4f3fd51e-a25d-11e5-9469-93d0d7f58baa.png)

**Comment by ctrueden** [comment]
> The :smile: is too much of a grin.

@Demon000 That is not something defined by the Gitter team. The Emoji are [used all over](http://www.emoji-cheat-sheet.com/) including on GitHub. I guess they could specifically override that one smiley but even so, that would be a separate issue to this one.

**Comment by lucaswerkmeister** [comment]
@ctrueden but neither that page nor GitHub defines a mapping from `:)` to `:smile:`, which is what this issue is all about.

**Comment by georgiel** [comment]
@demon000 please make that an issue,  I would like to support it

**Comment by asmeurer** [comment]
Or just 🙂

**Comment by smathy** [comment]
Bike-shedding alert!

Please feel free to take this "which emoji should `:)` be turned into" discussion elsewhere, **this** issue is about not turning `:)` into **ANY** emoji. Please stay on topic.

**Comment by mholt** [comment]
+1

Or at least ensure that if you convert :) to a picture, it is its own word. In one of my rooms we were talking about binding to a :port, and it rendered as :stuck_out_tongue_winking_eye:ort which is unacceptable.

**Comment by abhisekp** [comment]
I agree with @mholt 
If `:P` is rendered, it should be a word in itself. And for other smileys, they can be written as `:smile::smile::smile:`

**Comment by MichielBijl** [comment]
Any news on this?

**Comment by ashnur** [comment]
Apart from the fact that it's a terrible UX for a chat that's targeted at developers to automatically transform :<something> into an unrelated image, it's also a terrible desing simply because the damn smiley emoticons look like LAUGHTER instead of smileys. You can't put a smiley on gitter unless you use one of these: :relaxed: :innocent: :yum: :relieved: :smirk: :blush: :wink: :heart_eyes: 
Problem is, none of them are even close to a proper smiley that's just `:)` 
This is how a smiley emoticon should look like: http://usefulshortcuts.com/imgs/yahoo-smileys/1.gif

**Comment by dasilvacontin** [comment]
+1, it's sad that I cannot express `:)`.

**Comment by georgiel** [comment]
@dasilvacontin @ashnur @asmeurer 

Don't post about that here. We don't want it to get off-topic.

That's an unrealted issue, which has its own thread. https://github.com/gitterHQ/gitter/issues/1027

**Comment by georgiel** [comment]
#1027

**Comment by dasilvacontin** [comment]
> Don't post about that here. We don't want it to get off-topic. That's an unrelated issue, which has its own thread
> Only show emojis for `:emoji:`, not regular smileys

Sorry if I didn't make myself clear @georgiel, but I _do_ have an issue with `:)` (and others) getting automatically translated to an emoji (which I can't even preview until I hit send), and I would prefer if we could avoid such behaviour. And the fact that I don't feel like any emoji expresses `:)` makes it even worse.

To me, the fact that the emoji doesn't fully correlate with the smiley is a secondary issue. Because there isn't one I would be happy with anyways (at list from the pool of emojis that Gitter makes available to us).

**Comment by asmeurer** [comment]
A workaround I just discovered is to put a space after the colon, like : )

**Comment by Georgiel** [comment]
@dasilvacontin I realise that was part of what you said, but still, if it contains a seperate issue too in the comment it can make discussion go even further off-topic, now the issue has been split off to it's own thread #1027 .

**Comment by OvermindDL1** [comment]
Why is this closed?  The linked PR does not fix this issue of where things like :p or :) are getting translated into emoticons when they absolutely should not be.

**Comment by MadLittleMods** [comment]
Sorry, this was linked to the PR because it made references to people wanting better `:)` translation but thanks for pinging as the changes do not fix the issue originally raised.
